packages/model-viewer/src/features/scene-graph.ts

```diff
@@ -119,11 +120,20 @@ export const SceneGraphMixin = <T extends Constructor<ModelViewerElementBase>>(
       };
     }
 
-    async createTexture(uri: string, type: string = 'image/png'):
-        Promise<ModelViewerTexture|null> {
+    async createTexture(canvas: HTMLCanvasElement):
+        Promise<ModelViewerTexture|null>;
+    async createTexture(uri: string, type?: string):
+        Promise<ModelViewerTexture|null>;
+    async createTexture(
+        src: HTMLCanvasElement|string,
+        type: string = 'image/png'): Promise<ModelViewerTexture|null> {
       const currentGLTF = this[$currentGLTF];
-      const texture: Texture = await new Promise<Texture>(
-          (resolve) => this[$textureLoader].load(uri, resolve));
+
+      const texture: Texture = src instanceof HTMLCanvasElement ?
+          new CanvasTexture(src) :
```

Comment:
Will this behave as a video texture if the source canvas is updated every frame?

Reply:
No, unless `Image`'s `update` method is called on every frame.